server/src/main/java/org/elasticsearch/search/query/QuerySearchResult.java

```diff
@@ -220,6 +233,14 @@ public void releaseAggs() {
         }
     }
 
+    public void releaseAggregationContext() {
+        Releasables.close(toRelease);
+    }
+
+    public void addReleasable(Releasable releasable) {
+        toRelease.add(releasable);
```

Comment:
Are we sure `toRelease` is always assigned when this method is invoked?

Reply:
Currently? I'm pretty sure, but it's tricky. Right now, this class has basically three "modes" - data node, coordinating node, and null.  The data node constructor path correctly initializes `toRelease`, but neither of the other to do.  Coordinating mode could initialize it, and the associated ref counting, but we don't do anything with it there yet.  The null case probably shouldn't? I'm not entirely clear on why we need that path, but it's definitely used.  Seemed safer to throw (NPE) if we tried to add a releasable in a path we weren't expecting it, but I'm open to discussion if you'd rather do something else.